### PR TITLE
Add node sync period flag

### PR DESCRIPTION
**Pull request by deyuanix**
@erictune @dcarney

### CODE REVIEW

**Comment by bgrant0607** [comment]
How does this relate to #3498?

**Comment by ddysher** [comment]
This is to reduce calls to cloudprovider.  #3643

File: docs/man/kube-controller-manager.1.md [hunk 1]
```diff
@@ -54,11 +54,14 @@ The kube-controller-manager has several options.
...
 **--machines**=[]
-        List of machines to schedule onto, comma separated.
+    List of machines to schedule onto, comma separated.
 
 **--master**=""
 	The address of the Kubernetes API server.
 
+**--node_sync_peroid**=10s
+    The period for syncing nodes from cloudprovider, in nanosecond.
+
```
Comment (smarterclayton) [inline docs/man/kube-controller-manager.1.md]: Doesn't look like nanoseconds :)

File: docs/node.md [hunk 2]
```diff
@@ -58,7 +58,8 @@ objects. It performs two major functions: cluster-wide node synchronization
 and single node life-cycle management.
 
 Node controller has a sync loop that creates/deletes `Node`s from Kubernetes
-based on all matching VM instances listed from cloud provider. If a new instance
+based on all matching VM instances listed from cloud provider. The sync period
+can be controlled via flag "--node_sync_peroid". If a new instance
```
Comment (smarterclayton) [inline docs/node.md]: Typo: Period

File: docs/man/kube-controller-manager.1.md (hunk 1 above)
Comment (ddysher) [inline docs/man/kube-controller-manager.1.md]: OH right, go will parse the flag...

File: cmd/kube-controller-manager/controller-manager.go [hunk 3]
```diff
@@ -49,6 +49,7 @@ var (
 	cloudProvider   = flag.String("cloud_provider", "", "The provider for cloud services.  Empty string for no provider.")
 	cloudConfigFile = flag.String("cloud_config", "", "The path to the cloud provider configuration file.  Empty string for no configuration file.")
 	minionRegexp    = flag.String("minion_regexp", "", "If non empty, and -cloud_provider is specified, a regular expression for matching minion VMs.")
+	nodeSyncPeriod  = flag.Duration("node_sync_period", 10*time.Second, "The period for syncing nodes from cloudprovider.")
```
Comment (erictune) [inline cmd/kube-controller-manager/controller-manager.go]: How about adding this to flag docstring:
`Longer periods will result in fewer calls to cloud provider, but may delay addition of new nodes to cluster.`

**Comment by erictune** [comment]
LGTM, assuming travis passes.

**Comment by ddysher** [comment]
Comment addressed

**Pull request merged**